onnxruntime/core/optimizer/graph_transformer_utils.cc

```diff
@@ -130,6 +131,7 @@ std::vector<std::unique_ptr<GraphTransformer>> GenerateTransformers(TransformerL
 
       std::unordered_set<std::string> cuda_execution_providers = {onnxruntime::kCudaExecutionProvider};
       transformers.emplace_back(onnxruntime::make_unique<AddGeluFusion>(cuda_execution_providers));
+      transformers.emplace_back(onnxruntime::make_unique<GeluApproximation>(cuda_execution_providers));
```

Comment:
I think we should either add a new opt level or replace the AddGeluFusion GPU kernel with GeluApproximation kernel. 
If you take the latter option, you don't need to register a transformer here. You just need to replace the GPU kernel. I have a PR [https://github.com/microsoft/onnxruntime/pull/2487](https://github.com/microsoft/onnxruntime/pull/2487) that change AddGeluFusion to have same schema as the FastGelu.

Reply:
After more discussion, here is a summary:
(1) It is nice to have a session option to enable or disable approximation. That option could enable or disable this transformer. However, we think it is not worth the effort to add it right now.
(2) Another option is we add this flag to a runtime configuration (to be built) that could saved with model. This feature is not available right now.

I will add this to level 2 transformer for now. If user does need non-approximated version, he/she could use our python script to generate an optimized BERT model, and disable online optimization in inference.

In the future, we could evaluate user requirements to see whether it need to add an option to enable or disable approximation.